src/main/java/life/qbic/projectwizard/control/SampleCounter.java

```diff
@@ -49,23 +47,28 @@ public SampleCounter(String project) {
     barcode = "";
     // barcodeID = 1;
     this.project = project;
-    knownAtypical.add(project+"_INFO");
+    knownAtypical.add(project + "_INFO");
   }
 
   // TODO later updates (after initialization)
   public void increment(Sample s) {
     String code = s.getCode();
-    String experiment = s.getExperimentIdentifierOrNull();
+    String experimentID = null;
+    try {
+      experimentID = s.getExperiment().getIdentifier().getIdentifier();
+    } catch (Exception e) {
+    }
```

Comment:
remember TODO nullcheck?
This is also fine but I would ask you to include some logging here so it is not completely silent and undetectable. And specify what exception you are expecting here.
```suggestion
    try {
      experimentID = s.getExperiment().getIdentifier().getIdentifier();
    } catch (/*TODO WhatIsTheExceptionType*/ e) {
        log.warn(/*TODO compose a sensible message*/);
    }
```